grails-app/views/stockMovement/addComment.gsp

```diff
@@ -29,9 +30,25 @@
 			<g:render template="summary" model="[stockMovement:stockMovement]" />
 			<div class="box">
 				<h2><warehouse:message code="default.add.label" args="[entityName]" /></h2>
-				<g:form action="${comment?.id ? 'updateComment' : 'saveComment'}">
-					<g:hiddenField name="id" value="${comment?.id}" />
+
+				<g:set var="formAction"  value="saveComment" />
+				<g:set var="isRequestRejected" value="${approvalStatus == StockMovementStatusCode.REJECTED}" />
+				<g:if test="${comment?.id}">
+					<g:set var="formAction"  value="updateComment" />
+				</g:if>
+				<g:elseif test="${isRequestRejected}">
+					<g:set var="formAction" value="updateStatus" />
+				</g:elseif>
+				<g:form action="${formAction}">
+					<g:if test="${isRequestRejected}">
+						<g:hiddenField name="status" value="${StockMovementStatusCode.REJECTED}" />
+						<g:hiddenField name="id" value="${stockMovement?.id}" />
+					</g:if>
+					<g:if test="${comment?.id}">
+						<g:hiddenField name="id" value="${comment?.id}" />
+					</g:if>
```

Comment:
I kinda don't understand what is going on here, could you be that kind and explain those cases?

Reply:
Because I am using this same template for both `updateComment` and `updateStatus` which both expect a different id parameter, for updateComment `id = commented`  and for updateStatus `id = stockMovementId` I needed to wrap them inside the if statements to avoid stacking of values since calling this code
```
<g:hiddenField name="id" value="${stockMovement?.id}" />
<g:hiddenField name="id" value="${comment?.id}" />
```

will result in `prams.id` to have a value of an Array, it doesn't redefine the value but appends to the existing one of both are called.